### PR TITLE
Refactored K8sPlugin test

### PR DESCRIPTION
**Pull request by nfode**
Forgot to refactor the `transformTest()` in the K8sPluginTest.

### CODE REVIEW

File: core/src/test/java/org/opentosca/toscana/plugins/k8s/K8sPluginTest.java [hunk 1]
```diff
@@ -41,11 +42,10 @@
     @Autowired
     TestTransformationContext testTransformationContext;
```
Comment (hnicke) [inline core/src/test/java/org/opentosca/toscana/plugins/k8s/K8sPluginTest.java]: Limit the scope of fields: `private` would be appropriate.

**Comment by hnicke** [comment]
Due to applied fix 1ef422d786296f2bb611de985ba7248ebb0c4791 the transformTest() fails.
Currently blocked by #103 and #104.

File: core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    @Test(expected = InvalidDockerAppException.class)
+    public void dockerAppMissing() throws FileNotFoundException, InvalidCsarException, InvalidDockerAppException {
+        Csar csar = testCsars.getCsar(TestCsars.CSAR_YAML_INVALID_DOCKERAPP_MISSING);
+        tServiceTemplate = csarParser.parse(csar);
+        plugin.getDockerApp(tServiceTemplate);
+    }
+
+    @Test
+    public void transformTest() throws Exception {
+        File artifactRootPath = new File(preferences.getDataDir(), "test-csar/transformations/testplatform_one/");
```
Comment (hnicke) [inline core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java]: Your test should not hard code the artifact path. When the repo implementation changes, your test breaks.
As your using the TestTransformationContext, I suggest you use:
```
@Autowired
TransformationDao transformationDao;
transformationDao.getContentDir()
```

File: core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        TransformationContext transformationContext = testTransformationContext.getContext(CSAR_YAML_VALID_DOCKER_SIMPLETASK, TestPlugins.PLATFORM1);
+        assertNotNull(transformationContext);
+        plugin.transform(transformationContext);
+
+        checkFilesInTransformationRoot(artifactRootPath);
+        checkDockerFilesInTransformationRoot(artifactRootPath);
+    }
+
+    private void checkDockerFilesInTransformationRoot(File artifactRootPath) {
```
Comment (hnicke) [inline core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java]: Look, this is a perfect example: Yesterday, the TransformationDao implementation changed.
Transformation Artifacts (not the resulting archive, but the single files) are now copied to the subdir /contents in the transformationRootDir. If you need the paths, please use transformationDao.getContentDir() or transformationDao.getRootDir() (here you will find the .zip)

File: core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        List<String> result = new LinkedList<>();
+        result.clear();
+        for (String s : transformationDockerFilesPath.list()) {
+            result.add(s);
+        }
+        assertTrue(expectedDockerPathFiles.containsAll(result));
+    }
+
+    private void checkFilesInTransformationRoot(File artifactRootPath) {
+        List<String> expectedTransformationRoot = Lists.newArrayList("simple-task-app", "simple-task-app_resource.yaml", "Readme.md", "test-csar-testplatform_one.log");
```
Comment (hnicke) [inline core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java]: I would completey change the philosophy of the test: I suggest you mock the PluginFileAccess and verify that the methods got called. E.g., verify that `accessMopluginFileAccess.copy("simple-task-app")` got called.

File: core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java (hunk 2 above)
Comment (nfode) [reply core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java]: So you suggest changing the TestTransformationContext class?

File: core/pom.xml [hunk 5]
```diff
@@ -31,6 +31,12 @@
 
     <dependencies>
         <dependency>
+            <groupId>org.mockito</groupId>
```
Comment (c-mueller) [inline core/pom.xml]: Is this necessary? Doesn't spring boot test ship with Mockito?

File: docs/dev/api/api_samples.md [hunk 6]
```diff
@@ -79,13 +79,13 @@
 	"_embedded": {
 		"csar": [
 			{
-				"name": "k8s-cluster",
+				"name": kubernetes,
 				"_links": {
 					"self": {
-						"href": "http://localhost:8080/csars/k8s-cluster"
+						"href": kubernetes
```
Comment (c-mueller) [inline docs/dev/api/api_samples.md]: Hmm this is not the proper value.

File: core/pom.xml (hunk 5 above)
Comment (nfode) [reply core/pom.xml]: I have to retest but the version shipped with spring boot was to old to mock final classes.

File: docs/dev/api/api_samples.md (hunk 6 above)
Comment (nfode) [reply docs/dev/api/api_samples.md]: Ou something with refactoring went wrong.

File: core/pom.xml (hunk 5 above)
Comment (nfode) [reply core/pom.xml]: Yep I was right, the newes version is needed to mock `final` classes. In my case the `TransformationContext`

**Pull request merged**